### ISSUE TITLE
Rectangle rounded

### ISSUE THREAD

**Issue by djigit0s** [jorisschellekens/borb#87]
Hi. Is it possible to make the corners of the table and the rectangle rounded?

**Comment by djigit0s** [comment]
if anyone needs
```
def rectangle_corners(bounding_box: Rectangle, radius: int = 10):
    """
    This function returns the coordinates for a circle that fits in the given bounding box
    """
    points = []

    point1 = (bounding_box.x + radius, bounding_box.y)
    point2 = (bounding_box.x + bounding_box.width - radius, bounding_box.y)
    point3 = (bounding_box.x + bounding_box.width, bounding_box.y + radius)
    point4 = (bounding_box.x + bounding_box.width,
              bounding_box.y + bounding_box.height - radius)
    point5 = (bounding_box.x + bounding_box.width -
              radius, bounding_box.height)
    point6 = (bounding_box.x + radius, bounding_box.height)
    point7 = (bounding_box.x, bounding_box.height - radius)
    point8 = (bounding_box.x, bounding_box.y + radius)

    angle2 = []
    for k in range(90, 180):
        x = Decimal(math.sin(math.radians(k))) * radius
        y = Decimal(math.cos(math.radians(k))) * radius
        angle2.append((bounding_box.width - radius +
                      x, bounding_box.y + radius + y))
    angle3 = []
    for k in range(0, 90):
        x = Decimal(math.sin(math.radians(k))) * radius
        y = Decimal(math.cos(math.radians(k))) * radius
        angle3.append((bounding_box.width - radius + x,
                      bounding_box.height - radius + y))
    angle4 = []
    for k in range(270, 360):
        x = Decimal(math.sin(math.radians(k))) * radius
        y = Decimal(math.cos(math.radians(k))) * radius
        angle4.append((bounding_box.x + radius + x,
                      bounding_box.height - radius + y))
    angle1 = []
    for k in range(180, 270):
        x = Decimal(math.sin(math.radians(k))) * radius
        y = Decimal(math.cos(math.radians(k))) * radius
        angle1.append((bounding_box.x + radius + x,
                      bounding_box.y + radius + y))

    points = [
        point1,
        point2,
        *angle2[::-1],
        point3,
        point4,
        *angle3[::-1],
        point5,
        point6,
        *angle4[::-1],
        point7,
        point8,
        *angle1[::-1],
        point1
        # repeat first point to explicitly close shape
    ]

    return points
```

**Comment by jorisschellekens** [comment]
It is currently not possible to round the borders of `LayoutElement` objects.

I appreciate your work, but I think more effort will be needed to handle borders if `borb` needs to support both rounded borders and dashed borders.

I will keep your request in mind when/if I start development on these particular features.

Kind regards,
Joris Schellekens

**Comment by jorisschellekens** [comment]
I've implemented `border_radius` on all 4 corners of any `LayoutElement`.

I've added tests to showcase the new behaviour and it's options.

I also changed the code to draw the background to take into account the possibly rounded border.

Should be available in `v2.0.23`.
